UnityProject/Assets/Scripts/Systems/Antagonists/AntagData.cs

```diff
@@ -150,5 +224,53 @@ public static Objective PickRandomObjective(ref List<Objective> objectives, bool
 			}
 			return chosenObjective;
 		}
+
+		public Antagonist FromIndexAntag(short index)
```

Comment:
same here. I find index to be unreliable. Add a unique ID to every antag we spawn with the ``Guid.NewGuid()``. Same for objectives.

You can then store them in a dictionary and use the ID as key to get them later.